src/models/connectionCredentials.ts

```diff
@@ -43,36 +43,36 @@ export class ConnectionCredentials implements IConnectionCredentials {
      */
     public static createConnectionDetails(credentials: IConnectionCredentials): ConnectionDetails {
         let details: ConnectionDetails = new ConnectionDetails();
-        details.serverName = credentials.server;
-        if (credentials.port && details.serverName.indexOf(',') === -1) {
+        details['serverName'] = credentials.server;
```

Comment:
This should be `details.options['serverName']`. You changed the class to add a sub-property, but since you can assign fields using `['xyz']` Typescript isn't complaining.